File: src/main/java/io/pageobject/generator/locator/MultiElementLocatorPartGenerator.java
package io.pageobject.generator.locator;

import io.pageobject.generator.GeneratorContext;
import org.jsoup.nodes.Element;

public class MultiElementLocatorPartGenerator extends AbstractLocatorPartGenerator {

    public MultiElementLocatorPartGenerator(Element element, LocatorSources... locatorCandidates) {
        super(element, locatorCandidates);
    }

    @Override
    public LocatorPart onlyPart(GeneratorContext context) {
        return generateLastPart(context, ELEMENT_ARRAY_FINDER, 0);
    }

    @Override
    public LocatorPart lastPart(GeneratorContext context, int index) {
        return generateLastPart(context, ELEMENT_ARRAY_FINDER_IN_REPEATER, index);
    }
}

File: lib/locator/multi-element-locator-part-generator.js
"use strict";

const AbstractLocatorPartGenerator = require('./abstract-locator-part-generator');

class MultiElementLocatorPartGenerator extends AbstractLocatorPartGenerator {

  constructor(element, ...locatorCandidates) {
    super(element, locatorCandidates);
  }

  onlyPart(context) {
    return super.generateLastPart(context, super.elementArrayFinder, 0);
  }

  lastPart(context, index) {
    return super.generateLastPart(context, super.elementArrayFinderInRepeater, index);
  }

}

module.exports = MultiElementLocatorPartGenerator;